core/src/main/java/com/netflix/conductor/core/events/EventProcessor.java

```diff
@@ -1,235 +1,247 @@
+                int i = 0;
+                for (Action action : eventHandler.getActions()) {
+                    String id = msg.getId() + "_" + i++;
+                    EventExecution eventExecution = new EventExecution(id, msg.getId());
+                    eventExecution.setCreated(System.currentTimeMillis());
+                    eventExecution.setEvent(eventHandler.getEvent());
+                    eventExecution.setName(eventHandler.getName());
+                    eventExecution.setAction(action.getAction());
+                    eventExecution.setStatus(Status.IN_PROGRESS);
+                    if (executionService.addEventExecution(eventExecution)) {
+                        futuresList.add(CompletableFuture.supplyAsync(() -> execute(eventExecution, action, payload), executorService));
+                    } else {
+                        logger.warn("Duplicate delivery/execution of message: {}", msg.getId());
+                    }
+                }
+
+                CompletableFuture<List<Pair<EventExecution, Boolean>>> allFutures = CompletableFutures.allAsList(futuresList);
+                allFutures.whenComplete((result, error) -> result.forEach(eventExecutionBooleanPair -> {
+                    if (error != null || !eventExecutionBooleanPair.getValue()) {
+                        executionService.removeEventExecution(eventExecutionBooleanPair.getKey());
+                        failedExecutions.add(eventExecutionBooleanPair.getKey());
+                    } else {
+                        executionService.updateEventExecution(eventExecutionBooleanPair.getKey());
+                    }
+                })).get();
+            }
+
+            if (failedExecutions.isEmpty()) {
+                queue.ack(Collections.singletonList(msg));
+            } else if (queue.rePublishIfNoAck()) {
+                // re-submit this message to the queue, to be retried later
+                // This is needed for queues with no unack timeout, since messages are removed from the queue
+                queue.publish(Collections.singletonList(msg));
+            }
+
+        } catch (Exception e) {
+            logger.error("Error handling message: {} on queue:{}", msg, queue.getName(), e);
+        }
+    }
+
+    @SuppressWarnings("Guava")
+    @VisibleForTesting
+    Pair<EventExecution, Boolean> execute(EventExecution eventExecution, Action action, String payload) {
+        try {
+            String methodName = "executeEventAction";
+            String description = String.format("Executing action: %s for event: %s with messageId: %s with payload: %s", action.getAction(), eventExecution.getId(), eventExecution.getMessageId(), payload);
+            logger.debug(description);
+
+            Predicate<Throwable> filterException = throwableException -> {
+                if (throwableException != null) {
+                    return !(throwableException.getCause() instanceof UnsupportedOperationException);
```

Comment:
why checking the cause? does RetryUtil wrap the cause with another exception?

Reply:
yes, RetryUtil wraps it in ExecutionException